legendgram/legendgram.py

```diff
@@ -1,49 +1,64 @@
 from .util import make_location as _make_location
 import numpy as np
-
 from matplotlib.colors import Colormap
+import matplotlib.pyplot as plt
 from palettable.palette import Palette
 
 
-def legendgram(f, ax, y, breaks, pal, bins=50, clip=None,
+def legendgram(y, breaks=None, pal=None, bins=50, clip=None,
                loc = 'lower left', legend_size=(.27,.2),
-               frameon=False, tick_params = None):
+               frameon=False, tick_params = None, f=None, ax=None):
     '''
     Add a histogram in a choropleth with colors aligned with map
     ...
 
     Arguments
     ---------
-    f           : Figure
-    ax          : AxesSubplot
     y           : ndarray/Series
                   Values to map
-    breaks      : list
+    breaks      : list or int
+                  [Optional. Default=ten evenly-spaced percentiles from the 1st to the 99th]
                   Sequence with breaks for each class (i.e. boundary values
-                  for colors)
-    pal         : palettable colormap or matplotlib colormap
+                  for colors). If an integer is supplied, this is used as the number of
+                  evenly-spaced percentiles to use in the discretization 
+    pal         : palettable colormap, matplotlib colormap, or str
+                  palette to use to construct the legendgram. (default: None)
     clip        : tuple
                   [Optional. Default=None] If a tuple, clips the X
                   axis of the histogram to the bounds provided.
-    loc         :   string or int
-                    valid legend location like that used in matplotlib.pyplot.legend
+    loc         : str or int
+                  valid legend location like that used in matplotlib.pyplot.legend
     legend_size : tuple
                   tuple of floats between 0 and 1 describing the (width,height)
                   of the legend relative to the original frame.
     frameon     : bool (default: False)
                   whether to add a frame to the legendgram
     tick_params : keyword dictionary
                   options to control how the histogram axis gets ticked/labelled.
+    f           : Figuyre
+    ax          : AxesSubplot
 
     Returns
     -------
     axis containing the legendgram.
     '''
+    if f is None:
+        f = plt.gcf()
+    if ax is None:
+        ax = plt.gca()
+    if pal is None:
```

Comment:
Maybe I am missing something here, but why do you use None and an if clause for a default? You could just directly assign 'viridis' to the parameter in the api. This way users also see how the input out to look like?

Reply:
Just curious maybe I can learn something :)

Reply:
Good question! I had intended to somehow detect the current matplotlib palette and use that if `pal=None`. But, I couldn't quite figure out how to do that. So, I deleted the exploration in this `if` branch & simply used `pal = 'viridis'`. 

Ideally, you'd use `pal=None` and `if pal is None`, somehow check what colormap the user is using currently. I don't quite know how to do that.

Reply:
Ok, yeah that sounds great! 
do you think something like 
'''
fig, ax = plt.subplots()
im = ax.imshow(a, cmap="viridis")

cm = im.get_cmap()
print(cm.name)
"""

through

['Matplotlib.cm.get_cmap()'](https://matplotlib.org/3.2.1/api/cm_api.html) could work?

Happy to leave this as is and deal with it later. Maybe add a note, so others can keep finding solutions? :)

Reply:
Ah, well that's simole... how I missed it idk.  

No reason to merge an incomplete if that works! I'll take a look tomorrow.

Reply:
@ljwolf Is this still under consideration? I put together a [small gist](https://gist.github.com/jGaboardi/9c6f9b01f36c6a6c848538a4fc67235a) demonstrating a helper function built on top of @slumnitz's idea. Basically, it uses the most recent colormap of an image, if the axis has any associated images. Otherwise, it defaults to viridis.

Reply:
Yes, totally! push on top of this so that it updates!